### DISCUSSION TITLE
Cannot insert after soft delete values in prisma ORM

### DISCUSSION THREAD

**Discussion by kmihiranga** [prisma/prisma#15379] [Q&A] [answered]
I just need an idea about this. I have created a table with a unique column named `person_name`. In this table I use prisma v2 soft delete method. In here just update the deleted column as true. 

But when I try to insert the same `person_name` after a soft delete, it returns an error calling Unique constraint failed. 

Can anybody give any idea to skip this issue. My requirement is to make `person_name` as a unique column. But when I delete the `person_name` I should be able to insert the same `person_name` again.

**Comment by austincrim** [comment, marked as answer]
Hey @kmihiranga 👋,

I think the best way forward is to [customize the generated migration file](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/include-unsupported-database-features#customize-a-migration-to-include-an-unsupported-feature)and add a `WHERE` clause to the unique constraint, so it only applies where the `deleted` column is `false`. Something like [this](https://learn.microsoft.com/en-us/answers/questions/134187/unique-constraint-on-multiple-columns-and-soft-del.html).
